indexer/feature_altitude.cpp

```diff
@@ -0,0 +1,58 @@
+#include "indexer/feature_altitude.hpp"
+
+#include "base/bits.hpp"
+
+namespace feature
+{
+void Altitudes::PrepareSerializationData(TAltitude minAltitude, vector<uint32_t> & deviations) const
+{
+  CHECK(!m_altitudes.empty(), ());
+
+  deviations.clear();
+  uint32_t const firstPntDeviation =
+      bits::ZigZagEncode(static_cast<int32_t>(m_altitudes[0]) -
+      static_cast<int32_t>(minAltitude)) + 1 /* making it more than zero */;
+  CHECK_LESS(0, firstPntDeviation, ());
+  deviations.push_back(firstPntDeviation);
+  for (size_t i = 1; i < m_altitudes.size(); ++i)
+  {
+    uint32_t const nextPntDeviation = bits::ZigZagEncode(static_cast<int32_t>(m_altitudes[i]) -
+        static_cast<int32_t>(m_altitudes[i - 1])) + 1 /* making it more than zero */;
+    CHECK_LESS(0, nextPntDeviation, ());
+    deviations.push_back(nextPntDeviation);
+  }
+}
+
+bool Altitudes::FillAltitudesByDeserializedDate(TAltitude minAltitude, vector<uint32_t> const & deviations)
+{
+  m_altitudes.clear();
+  if (deviations.size() == 0)
+  {
+    ASSERT(false, ());
+    return false;
+  }
+
+  m_altitudes.resize(deviations.size());
```

Comment:
m_altitudes should be empty if an error happened in FillAltitudesByDeserializedDate. I moved m_altitudes.resize(deviations.size()); above but left m_altitudes.clear(); in case of ASSERTs.